lightning-block-sync/src/convert.rs

```diff
@@ -298,14 +302,14 @@ pub(crate) mod tests {
 		fn from(data: BlockHeaderData) -> Self {
 			let BlockHeaderData { chainwork, height, header } = data;
 			serde_json::json!({
-				"chainwork": chainwork.to_string()["0x".len()..],
+				"chainwork": chainwork.to_be_bytes().as_hex().to_string(),
 				"height": height,
-				"version": header.version,
-				"merkleroot": header.merkle_root.to_hex(),
+				"version": header.version.to_consensus(),
+				"merkleroot": header.merkle_root.to_string(),
 				"time": header.time,
 				"nonce": header.nonce,
-				"bits": header.bits.to_hex(),
-				"previousblockhash": header.prev_blockhash.to_hex(),
+				"bits": header.bits.to_consensus().to_be_bytes().as_hex().to_string(),
```

Comment:
`format!("{08x}", header.bits.to_consensus())` would convey the meaning better but your code is actually faster because std formatting is over-complicated. I've opened an issue to support hex directly on `CompactTarget` but that will still be slower than your code.